scripts/tempesta.sh

```diff
@@ -234,24 +234,24 @@ prepare_db_directory()
 	rm -f /opt/tempesta/db/*.tdb;
 }
 
-start_tempesta_and_check()
+start_tempesta_and_check_state()
 {
-	# If 'net.tempesta.state' entry exists but Tempesta start process has
-	# been failed for some reason (e.g. configuration parsing error), then
-	# 'sysctl' does not indicate any problems and exits with zero code;
-	# so, stderr may return 'sysctl: setting key "net.tempesta.state"'
-	# followed by some error message or just empty string.
-	# The most reliable way to check Tempesta status is to check
-	# net.tempesta.state and if it is "start" check dmesg.
-	err=$(sysctl -w net.tempesta.state=start 2>&1 1>/dev/null)
+	local _err
+
+	_err=$(sysctl -w net.tempesta.state=start 2>&1 1>/dev/null)
 	TFW_STATE=$(sysctl net.tempesta.state 2> /dev/null)
-	TFW_STATE=${TFW_STATE##* }
+	TFW_STATE=${TFW_STATE##net.tempesta.state = }
 
-	if [[ ${TFW_STATE} != "start" ]] || [[ -z "`check_dmesg 'Tempesta FW is ready'`" ]]; then
-		echo $err
+	remove_tmp_conf
```

Comment:
Pay attention on this place now i remove tmp conf before check state (so it will be removed in case of error also

Reply:
In my understanding we're safe here: `tfw_start()` is called in the process context of the sysctl write, so the file was read at this moment.